File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.js
class Solution {
    /**
     * @param {string} s
     * @return {number}
     */
    lengthOfLongestSubstring(s) {

        let charSet = new Set();

        let start = 0
        let longestSub = 0

        for (let end = 0; end < s.length; end++){
            while (charSet.has(s[end])){
                charSet.delete(s[start]);
                start += 1
            }
            
            charSet.add(s[end]);
            longestSub = Math.max(longestSub, end - start + 1)
            
        }
        return longestSub
    }

}

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:

        start = 0
        end = 0
        max_profit = 0

        for i in prices:

            while end < len(prices):
                if prices[start] < prices[end] :
                    max_profit = max(max_profit, prices[end] - prices[start])
                else:
                    start = end
                end += 1
        return max_profit

            


        